### PR TITLE
WP-424 app route plugin

### PR DESCRIPTION
**Pull request by mmcgahan**
Main changes involve consolidating the 'app route' into a single Hapi plugin

1. `src/routes/appRouteHandler.js` -> `src/plugins/app-route/handler.js`
2. `src/routes/appRoute.js` -> `src/plugins/app-route/route.js`
3. the server app rendering functions created in the server-app build process are now passed to the app route as a plugin option to the `getPlugins()` function. Previously, they were passed in to the `getRoutes()` function. Minor difference.
4. the entire plugin is covered by flowtype
5. `renderRequestMap` renamed to `languageRenderers`, which is a bit more clear about what is contained in the object
6. app route test moved from general routes module test to a plugin-route-specific test file.

### CODE REVIEW

File: tests/integration/server.int.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import { mockConfig } from '../mocks';
 import start from '../../src/server';
-import * as appRouteHandler from '../../src/routes/appRouteHandler';
+import * as appRouteHandler from '../../src/plugins/app-route/handler';
```
Comment (willh-meetup) [inline tests/integration/server.int.js]: Another thing probably outside the scope of this...

Might be nice to have an integration test for the language redirect behavior.

File: src/plugins/app-route/route.js [hunk 2]
```diff
@@ -41,12 +37,8 @@ export const onPreResponse = {
  * - `state`: skip cookie validation because cookies can be set by other services
  *   on the same domain and may not conform to the validation rules that are
  *   used by Hapi
- *
- * @param {Object} renderRequestMap a map of 'localeCode' strings to request-
- *   rendering functions that return an HTML string
- * @return {Object} a Hapi route configuration object
  */
-const getApplicationRoute = renderRequestMap => ({
+export default (languageRenderers: { [string]: LanguageRenderer$ }) => ({
```
Comment (willh-meetup) [inline src/plugins/app-route/route.js]: 👍  `languageRenderers` is a better name than `renderRequestMap`

File: src/plugins/app-route/handler.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+// @flow
```
Comment (willh-meetup) [inline src/plugins/app-route/handler.js]: Might be worth adding a comment to this file explaining exactly what it's handling (language redirects and session tracking).

File: src/plugins/app-route/handler.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+import url from 'url';
+
+export default (languageRenderers: { [string]: LanguageRenderer$ }) => (
+	request: HapiRequest,
+	reply: HapiReply
+) => {
+	const pathname = request.getPrefixedPath();
```
Comment (mmcgahan) [inline src/plugins/app-route/handler.js]: Makes sense - I'll do that in a separate branch and merge straight to `next` before this PR merges

**Pull request merged**